BerkeleyHCI/PolymorphicBlocks
# BLDC driver and SWD debugger spin

As the title implies.

Design changes:
- Layout not included (for an external project).
- BLDC: 
  - Remove the extraneous stuff from the prior omnibus run
  - Replace micro with a Feather nRF52840 dev board.
  - Add hall sensors, motor voltage sense, motor current sense
  - TEST POINTS EVERYWHERE
  - Move PowerOutConnector to fcml (prior in bldc for the buck-boost converter)
- SWD:
  - Change to DAPLink pinning
  - Add target reset switch and pull-up (since this acts as bootloader entry on DAPLink)
  - Use CSTNE ceramic resonator, which should be smaller than a crystal

Infrastructural changes:
- Adds ceramic resonators as an option for for oscillator references.
- Adds the more abstract OscillatorReference that encapsulates both crystals and ceramic resonators and makes them interchangeable.
- Add CSTNE series ceramic resonators (well, only the 8MHz low tolerance one)
- Crystal port now defines `xtal_in` and `xtal_out` - the distinction for which apparently can't be ignored for ceramic resonators
- Add pull-up and pull-down array, sized from port array connections
- Change `__init__` to `contents` in some micro defs
- Refactor nRF52840 to be mixin compliant including for the Feather socketed dev board
